src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -1783,7 +1800,7 @@ ExtractParametersFromParamListInfo(ParamListInfo paramListInfo, Oid **parameterT
 		 * the master and worker nodes. Therefore, the worker nodes can
 		 * infer the correct oid.
 		 */
-		if (parameterData->ptype >= FirstNormalObjectId)
+		if (parameterData->ptype >= FirstNormalObjectId && !isLocalExecution)
```

Comment:
what happens if you always assign `(*parameterTypes)[parameterIndex] = parameterData->ptype;`

Reply:
I mean, what is the difference in passing `InvalidOid` to the remote nodes vs `wrongOid`. I'm not saying we should do it, just curious.

If postgres is ignoring the type, we might do that.

Reply:
@onderkalaci, with that we dont get the same behavior:

```
 -- test parameterized inserts
 PREPARE prepared_insert(varchar(20)) AS
 	INSERT INTO router_executor_table VALUES (1, $1, $2);
 EXECUTE prepared_insert('comment-1', '(1, 10)');
+ERROR:  cache lookup failed for type 22949
```

https://circleci.com/gh/citusdata/citus/51604